File: P1 2016.2/Questao(2)/java/entities/produtos/Produto.java
package entities.produtos;

public abstract class Produto {
    private double preco;
    private int id;

    public Produto(double preco, int id) {
        this.preco = preco;
        this.id = id;
    }

    public double getPreco() {
        return preco;
    }

    public void setPreco(double preco) {
        this.preco = preco;
    }

    public int getId() {
        return id;
    }

    @Override
    public boolean equals(Object p) {
        if (((Produto) p).getId() == getId()) {
            return true;
        }
        return false;
    }
}

File: P1 2016.2/Questao(2)/python/entities/produtos/Produto.py
class Produto():

    def __init__(self, preco: float, id: int) -> None:
        self.preco = preco
        self.id = id

    def __eq__(self, other) -> bool:
        if isinstance(other, Produto):
            return self.id == other.id
        return False

    def __str__(self) -> str:
        return f"Preco: {self.preco}, ID:{self.id}"
